spec/defines/dns__record__txt_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'spec_helper'
+
+describe 'dns::record::txt', :type => :define do
+  let(:title) { 'txttest' }
+  let(:facts) { { :concat_basedir => '/tmp' } }
+
+  context 'passing a simple string should result in a quoted string' do
+    let :params do {
+        :host => 'txttest',
+        :zone => 'example.com',
+        :data => 'testing',
+    } end
+    it { should_not raise_error }
+    it { should contain_concat__fragment('db.example.com.txttest,TXT,example.com.record').with_content(/^txttest\s+IN\s+TXT\s+"testing"$/) }
+  end
+
+  context 'passing a string that includes a quote should result in the quote being escaped' do
+    let :params do {
+        :host => 'txttest',
+        :zone => 'example.com',
+        :data => 'this is a "test"',
+    } end
+    it { should_not raise_error }
+    it { should contain_concat__fragment('db.example.com.txttest,TXT,example.com.record').with_content(/^txttest\s+IN\s+TXT\s+"this is a \\"test\\""$/) }
+  end
+
+  context 'passing a long string should result in multiple quoted strings' do
+    let :params do {
+        :host => 'txttest',
+        :zone => 'example.com',
+        :data => 'this is a ' + 'very '*60 + 'long test',
+    } end
+    it { should_not raise_error }
+    it { should contain_concat__fragment('db.example.com.txttest,TXT,example.com.record').with_content(/^txttest\s+IN\s+TXT\s+"this is a very.*" ".*very long test"$/) }
```

Comment:
Can you add a note to the TXT/SPF puppet defines to indicate that this escaping is done for you so the user doesn't think they have to do it?